File: src/c01/ex3.scala
def larger_sq (x: Int, y: Int, z: Int): Int = {
  if (x < y && x < z) {
    sum_sq(y,z)
  } else if (y < x && y < z) {
    sum_sq(x,z)
  } else {
    sum_sq(x,y)
  }
}

def sum_sq (x: Int, y: Int): Int = {
  x * x + y * y
}

val shouldEqual_13 = larger_sq(1,2,3) == 13
val shouldEqual_25 = larger_sq(4,2,3) == 25
val shouldEqual_41 = larger_sq(4,5,3) == 41
val shouldEqual_20 = larger_sq(4,2,1) == 20

File: src/c01/ex3.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';
use feature qw(say);
use List::Util qw(min);


sub larger_sq {
    my ($x,$y,$z) = @_;
    my $lowest = min($x,$y,$z);
    if ($lowest == $x) {
        return sum_sq ($y,$z);
    } elsif ($lowest == $y) {
        return sum_sq($x,$z);
    } else {
        return sum_sq($x,$y);
    }
}

sub sum_sq {
    my ($x,$y) = @_;
    ($x * $x) + ($y * $y);
}


say(larger_sq(1,2,3));
say(larger_sq(4,2,3));
say(larger_sq(4,5,3));
say(larger_sq(4,1,3));
